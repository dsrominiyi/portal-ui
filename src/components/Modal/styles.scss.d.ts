export const footerActions: string;
export const modalOverlay: string;
export const modal: string;
export const visible: string;
export const body: string;
export const header: string;
export const footer: string;
export const closeIcon: string;
export const overlayEnter: string;
export const overlayEnterActive: string;
export const overlayExit: string;
export const overlayExitActive: string;
