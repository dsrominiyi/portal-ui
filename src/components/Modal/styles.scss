@import '../../style/vars';

$overlay-transition-ms: 300ms;
$dialog-transition-ms: 300ms;

.footer-actions {
  display: flex;
  justify-content: flex-end;

  > * {
    margin-right: 0.75rem;
  }

  > :last-child {
    margin-right: 0;
  }
}

.modal-overlay {
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  padding: 50px 30px;
  background-color: transparentize($brand-black, 0.4);
  z-index: 1000;
}

.modal {
  position: relative;
  display: flex;
  flex-direction: column;
  max-width: 85%;
  max-height: 100%;
  background-color: $brand-white;
  color: $brand-heavy-thunder;
  opacity: 0;
  transform: translateY(-100px);
  transition-duration: $dialog-transition-ms;
  transition-property: opacity, transform;

  &.visible {
    opacity: 1;
    transform: translateY(0);
  }

  .body {
    padding: 30px;
    overflow-y: auto;

    > * {
      margin: 20px 0;
    }

    > :first-child {
      margin-top: 0;
    }

    > :last-child {
      margin-bottom: 0;
    }
  }

  .header,
  .footer {
    padding: 15px 30px;
  }

  .header {
    border-bottom: 1px solid $brand-off-white;
    font-size: 1.75rem;
    font-weight: bold;
  }

  .footer {
    border-top: 1px solid $brand-off-white;
  }

  .closeIcon {
    position: absolute;
    display: flex;
    top: 8px;
    right: 8px;
    cursor: pointer;

    svg {
      width: 14px;
      height: 14px;

      > * {
        stroke: $brand-stone-grey;
      }

      &:hover {
        > * {
          stroke: $brand-danger;
        }
      }
    }
  }
}

.overlay-enter {
  opacity: 0;
}

.overlay-enter-active {
  opacity: 1;
  transition: opacity $overlay-transition-ms;
}

.overlay-exit {
  opacity: 1;
}

.overlay-exit-active {
  opacity: 0;
  transition: opacity $overlay-transition-ms;
}
