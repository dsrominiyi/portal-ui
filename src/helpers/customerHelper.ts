import { dateFieldValuesToDateString, dateStringToDateFieldValues } from './utilFuncs';
import { predefinedRules, ValidationRules } from './formValidator';

export const companyFormValidationRules: ValidationRules = {
  companyName: { ...predefinedRules.entityName, fieldLabel: 'company name', required: true },
  tradingName: { ...predefinedRules.entityName, fieldLabel: 'trading name' },
  typeStatus: { fieldLabel: 'company legal status', invalidValues: ['Other'], required: true },
  businessClass: { fieldLabel: 'business class', required: true },
  tradingSinceDate: { fieldLabel: 'trading since', isDate: true },
  parentCompany: { fieldLabel: 'is this company a subsidiary', required: true },
  parentCompanyName: { ...predefinedRules.entityName, fieldLabel: 'parent company name' },
  registeredCoNumber: predefinedRules.registeredCoNumber,
  contactDetails: { fieldLabel: 'contact details', required: true },
  addresses: { fieldLabel: 'addresses', required: true }
};

export const isIndividualCustomer = (customer: Customer): customer is Individual => {
  return typeof (customer as Individual).name !== 'undefined';
};

export const getIndividualFullName = ({ title, forename, middleName, surname }: IndividualName) => {
  return `${title} ${forename}${middleName && ` ${middleName}`} ${surname}`;
};

export const buildCompanyFromFormData = (data: NewCompanyData): Company => ({
  ...data,
  parentCompany: data.parentCompany || false,
  tradingSinceDate: dateFieldValuesToDateString(data.tradingSinceDate)
});

export const buildFormDataFromCompany = (data: Company): NewCompanyData => ({
  ...data,
  tradingSinceDate: data.tradingSinceDate
    ? dateStringToDateFieldValues(data.tradingSinceDate)
    : { dd: '', mm: '', yyyy: '' }
});
