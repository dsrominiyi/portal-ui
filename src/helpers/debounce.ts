/**
 * Prevents an action from being called more that once every `wait` milliseconds
 * The initial triggering of the action will be delayed by `wait` milliseconds if `immediate` = false
 *
 * e.g. <button onClick={event => debounce(handleClick, 200)(event)} />
 */

let timeout: number | null;

const debounce = (action: (...args: any[]) => void, wait = 300, immediate = true) => (
  ...args: any[]
) => {
  const later = () => {
    timeout = null;
    if (!immediate) {
      action(...args);
    }
  };
  const callNow = immediate && !timeout;

  if (timeout) {
    clearTimeout(timeout);
  }

  timeout = (setTimeout as Window['setTimeout'])(later, wait);

  if (callNow) {
    action(...args);
  }
};

export default debounce;
