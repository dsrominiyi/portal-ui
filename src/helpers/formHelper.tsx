import React, { ReactNode } from 'react';

import QuestionBox from '../components/QuestionBox';
import TextField, { Props as TextFieldProps } from '../components/TextField';
import AbiSearch, { Props as AbiSearchProps } from '../components/AbiSearch';
import Dropdown, { DropdownOption, Props as DropdownProps } from '../components/Dropdown';
import MultiChoice, { Props as MultiChoiceProps } from '../components/MultiChoice';
import DateFields, { Props as DateFieldsProps } from '../forms/common/DateFields';
import Lookup from '../components/Lookup';

import { ValidationError } from './formValidator';

import formStyles from '../forms/styles.scss';

interface InputConfig {
  type: InputType;
  label: string;
  buttonLabel?: string;
  secondaryButtonLabel?: string;
  value?: FieldValue;
  codeListId?: string;
  options?: DropdownOption[] | string[];
  attributes?: { [attr: string]: string };
  onChange?:
    | TextFieldProps['onChange']
    | AbiSearchProps['onChange']
    | DropdownProps['onChange']
    | DateFieldsProps['onChange']
    | MultiChoiceProps['onClick'];
  secondaryAction?: () => void;
}

export const buildInput = (config: InputConfig, fieldError?: ValidationError) => {
  let input: ReactNode;

  switch (config.type) {
    case InputType.Text:
      input = (
        <TextField
          key="0"
          additionalClasses={[formStyles.formTextField]}
          label={config.label}
          value={config.value as string}
          isInvalid={fieldError?.showError}
          onChange={config.onChange as TextFieldProps['onChange']}
          attributes={config.attributes}
        />
      );
      break;
    case InputType.Lookup:
      input = (
        <Lookup
          key="0"
          label={config.label}
          buttonLabel={config.buttonLabel}
          secondaryButtonLabel={config.secondaryButtonLabel}
          value={config.value as string}
          isInvalid={fieldError?.showError}
          onChange={config.onChange as TextFieldProps['onChange']}
          secondaryAction={config.secondaryAction}
          attributes={config.attributes}
        />
      );
      break;
    case InputType.AbiSearch:
      input = (
        <AbiSearch
          key="0"
          label={config.label}
          codeListId={config.codeListId || ''}
          onChange={config.onChange as AbiSearchProps['onChange']}
          attributes={config.attributes}
        />
      );
      break;
    case InputType.Dropdown:
      input = (
        <Dropdown
          key="0"
          label={config.label}
          options={config.options!}
          selected={config.value as string}
          isInvalid={fieldError?.showError}
          onChange={config.onChange as DropdownProps['onChange']}
          attributes={config.attributes}
        />
      );
      break;
    case InputType.DateFields:
      input = (
        <DateFields
          key="0"
          values={config.value as DateFieldValues}
          onChange={config.onChange as DateFieldsProps['onChange']}
          testId={config.attributes && config.attributes['test-id']}
        />
      );
      break;
    case InputType.MultiChoice:
      input = (
        <MultiChoice
          key="0"
          options={config.options as string[]}
          selectedOption={config.value as string}
          onClick={config.onChange as MultiChoiceProps['onClick']}
          attributes={config.attributes}
        />
      );
      break;
  }

  return (
    <QuestionBox
      key={config.label}
      text={[InputType.DateFields, InputType.MultiChoice].includes(config.type) ? config.label : ''}
      hasError={fieldError?.showError}
      errorText={fieldError?.message}
      inputFields={[input]}
    />
  );
};
