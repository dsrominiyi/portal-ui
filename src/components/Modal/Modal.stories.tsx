import React, { FunctionComponent, ReactNode, useState, Fragment } from 'react';
import { storiesOf } from '@storybook/react';

import Link from '../Link';
import Modal from '.';

import { storyRenders as questionBoxStories } from '../QuestionBox/QuestionBox.stories';
import { storyRenders as buttonStories } from '../Button/Button.stories';

import { footerActions } from './styles.scss';

interface ShowModalProps {
  renderModal(props: any): ReactNode;
}

const stories = storiesOf('Modal', module).addParameters({
  info: { source: false, propTables: [Modal] }
});

export const render = (body: ReactNode[], header?: ReactNode, footer?: ReactNode) => (
  props: any = {}
) => (
  <Modal
    isVisible={typeof props.isVisible === 'undefined' ? true : props.isVisible}
    body={body}
    header={header}
    footer={footer}
    hideModal={props.hideModal || (() => {})}
  />
);

const ShowModal: FunctionComponent<ShowModalProps> = ({ renderModal }) => {
  const [isVisible, setVisible] = useState(true);
  const hideModal = () => setVisible(false);

  return (
    <Fragment>
      <Link label="Show modal" onClick={() => setVisible(true)} />
      {renderModal({ isVisible, hideModal })}
    </Fragment>
  );
};

const header = 'Modal Header';
const body = [
  questionBoxStories.withAffixedField({ key: 0 }),
  questionBoxStories.withDropdown({ key: 1 })
];
const longBody = [
  questionBoxStories.withAffixedField({ key: 0 }),
  questionBoxStories.withDropdown({ key: 1 }),
  questionBoxStories.withAffixedField({ key: 0 }),
  questionBoxStories.withDropdown({ key: 1 }),
  questionBoxStories.withAffixedField({ key: 0 }),
  questionBoxStories.withDropdown({ key: 1 }),
  questionBoxStories.withAffixedField({ key: 0 }),
  questionBoxStories.withDropdown({ key: 1 }),
  questionBoxStories.withAffixedField({ key: 0 }),
  questionBoxStories.withDropdown({ key: 1 }),
  questionBoxStories.withAffixedField({ key: 0 }),
  questionBoxStories.withDropdown({ key: 1 })
];
const footer = <div className={footerActions}>{buttonStories.default()}</div>;

export const storyRenders = {
  withHeaderAndFooter: render(body, header, footer),
  longBodyWithHeaderAndFooter: render(longBody, header, footer),
  noFooter: render(body, header),
  noHeaderOrFooter: render(body)
};

stories.add('With header & footer', () => (
  <ShowModal renderModal={storyRenders.withHeaderAndFooter} />
));
stories.add('No footer', () => <ShowModal renderModal={storyRenders.noFooter} />);
stories.add('No header or footer', () => <ShowModal renderModal={storyRenders.noHeaderOrFooter} />);
stories.add('Long body with footer and header', () => (
  <ShowModal renderModal={storyRenders.longBodyWithHeaderAndFooter} />
));
