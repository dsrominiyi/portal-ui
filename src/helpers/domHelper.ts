import debounce from './debounce';

export const preventBodyScroll = () => {
  const { scrollY } = window;
  document.body.style.position = 'fixed';
  document.body.style.top = document.body.style.top || `-${scrollY}px`;
};

export const allowBodyScroll = () => {
  const scrollY = document.body.style.top;
  document.body.style.position = '';
  document.body.style.top = '';
  return scrollY && window.scrollTo(0, -parseInt(scrollY, 10));
};

export const triggerOnEnter = (action: () => void, debounceWait?: number) => (
  event: React.KeyboardEvent<Element | HTMLElement>
) => {
  if (event.key === 'Enter') {
    debounce(action, debounceWait)();
  }
};
