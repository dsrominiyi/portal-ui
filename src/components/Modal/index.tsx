import React, { FunctionComponent, ReactNode, useEffect, useState, useRef, RefObject } from 'react';
import { Portal } from 'react-portal';
import { CSSTransition } from 'react-transition-group';
import classnames from 'classnames/bind';

import CloseIcon from 'uikit-icons/lib/Close';

import { preventBodyScroll, allowBodyScroll } from '../../helpers/domHelper';

import styles from './styles.scss';

interface Props {
  isVisible: boolean;
  body: ReactNode[];
  header?: ReactNode;
  footer?: ReactNode;
  hideModal(): void;
}

type MouseDownTargetState = [
  HTMLElement | undefined,
  React.Dispatch<React.SetStateAction<HTMLElement | undefined>>
];

const cx = classnames.bind(styles);

const otherModalsDetected = () => document.querySelectorAll(`.${styles.modal}`).length > 1;

// transition times should match those defined in the stylesheet
const overlayTransitionMs = 300;
const dialogTransitionMs = 300;

export const Modal: FunctionComponent<Props> = ({
  isVisible,
  body,
  header,
  footer,
  hideModal
}: Props) => {
  const [isDialogVisible, setDialogVisible] = useState(false);
  const [isOverlayVisible, setOverlayVisible] = useState(false);
  const [hasMounted, setHasMounted] = useState(false);
  const [mouseDownTarget, setMouseDownTarget]: MouseDownTargetState = useState();
  const dialogRef: RefObject<HTMLDivElement> = useRef(null);

  const detectMouseDownTarget = (event: React.MouseEvent<HTMLElement>) => {
    event.stopPropagation(); // prevents nested modals from dismissing their parents
    setMouseDownTarget(event.target as HTMLElement);
  };
  const hideOnOverlayClick = (event: React.MouseEvent<HTMLElement>) => {
    const isClickOnOverlay =
      mouseDownTarget &&
      !dialogRef.current?.contains(mouseDownTarget) &&
      !dialogRef.current?.contains(event.target as HTMLElement);
    return isClickOnOverlay && hideModal();
  };
  const hideOnEsc = (event: KeyboardEvent) => ['Esc', 'Escape'].includes(event.key) && hideModal();

  useEffect(() => {
    document.body.addEventListener('keyup', hideOnEsc);
    const cleanUp = () => document.body.removeEventListener('keyup', hideOnEsc);
    return cleanUp;
  }, []);

  useEffect(() => {
    if (isVisible) {
      preventBodyScroll();
      setOverlayVisible(true);
      setTimeout(() => setDialogVisible(true), overlayTransitionMs);
    } else {
      // ensure nested modals don't allow body scroll when mounting or on dismissal
      if (hasMounted && !otherModalsDetected()) {
        allowBodyScroll();
      }
      setDialogVisible(false);
      setTimeout(() => setOverlayVisible(false), dialogTransitionMs);
    }
    setHasMounted(true);
  }, [isVisible]);

  return (
    <Portal>
      <CSSTransition
        in={isOverlayVisible}
        timeout={overlayTransitionMs}
        classNames={{
          enter: styles.overlayEnter,
          enterActive: styles.overlayEnterActive,
          exit: styles.overlayExit,
          exitActive: styles.overlayExitActive
        }}
        mountOnEnter
        unmountOnExit
      >
        <div
          className={styles.modalOverlay}
          onMouseDown={detectMouseDownTarget}
          onMouseUp={hideOnOverlayClick}
        >
          <div
            ref={dialogRef}
            className={cx({ modal: true, visible: isDialogVisible })}
            onMouseDown={detectMouseDownTarget}
          >
            <div className={styles.closeIcon} onClick={hideModal}>
              <CloseIcon viewBox="0.5 0 20 20" />
            </div>
            {header && <div className={styles.header}>{header}</div>}
            <div className={styles.body}>{body}</div>
            {footer && <div className={styles.footer}>{footer}</div>}
          </div>
        </div>
      </CSSTransition>
    </Portal>
  );
};

Modal.displayName = 'Modal';
export default Modal;
