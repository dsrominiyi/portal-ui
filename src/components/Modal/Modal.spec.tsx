import { mount, ReactWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';

import { preventBodyScroll, allowBodyScroll } from '../../helpers/domHelper';

import { render as renderModal } from './Modal.stories';

jest.mock('../../helpers/domHelper');
jest.useFakeTimers();

let component: ReactWrapper;
const hideModal = jest.fn();
document.body.addEventListener = jest.fn();
document.body.removeEventListener = jest.fn();

describe('Modal', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    document.body.innerHTML = '';
    component = mount(renderModal(['body'], 'header', 'footer')({ isVisible: true, hideModal }));
  });

  test('visible Modal matches snapshot', () => {
    act(() => {
      jest.runAllTimers();
    });
    component.update();

    expect(component).toMatchSnapshot();
  });

  test('hidden Modal matches snapshot', () => {
    component.setProps({ isVisible: false });
    act(() => {
      jest.runAllTimers();
    });
    component.update();

    expect(component).toMatchSnapshot();
  });

  test('page body scroll is prevented when the modal is visible', () => {
    expect(preventBodyScroll).toHaveBeenCalled();
  });

  test('page body scroll is allowed when the modal is hidden', () => {
    component.setProps({ isVisible: false });

    expect(allowBodyScroll).toHaveBeenCalled();
  });

  test('a keyup event listener is added to document.body on mount', () => {
    expect(document.body.addEventListener).toHaveBeenCalledWith('keyup', expect.any(Function));
  });

  test('the keyup event listener calls hideModal if the escape key is pressed', () => {
    const keyboardEvent = { key: 'Esc' };
    const keyupListener = (document.body.addEventListener as jest.Mock).mock.calls[0][1];
    keyupListener(keyboardEvent);

    expect(hideModal).toHaveBeenCalled();
  });

  test('the keyup event listener is removed on unmount', () => {
    component.unmount();

    expect(document.body.removeEventListener).toHaveBeenCalledWith('keyup', expect.any(Function));
  });

  test('clicking the modal overlay calls hideModal', () => {
    const dialogDiv = (component.find('.modal').getElement() as any).ref.current;
    dialogDiv.contains = jest.fn(() => false);
    component.find('.modalOverlay').simulate('mousedown', { target: {} });
    component.find('.modalOverlay').simulate('mouseup', { target: {} });

    expect(hideModal).toHaveBeenCalled();
  });

  test('clicking the modal dialog then releasing the click on the overlay does not call hideModal', () => {
    const dialogDiv = (component.find('.modal').getElement() as any).ref.current;
    dialogDiv.contains = jest
      .fn()
      .mockReturnValueOnce(true)
      .mockReturnValueOnce(false);
    component.find('.modalOverlay').simulate('mousedown', { target: {} });
    component.find('.modalOverlay').simulate('mouseup', { target: {} });

    expect(hideModal).not.toHaveBeenCalled();
  });

  test('clicking the modal overlay calls hideModal if the ref is not available', () => {
    const dialogRef = (component.find('.modal').getElement() as any).ref;
    dialogRef.current = null;
    component.find('.modalOverlay').simulate('mousedown', { target: {} });
    component.find('.modalOverlay').simulate('mouseup', { target: {} });

    expect(hideModal).toHaveBeenCalled();
  });

  test('clicking the close icon calls hideModal', () => {
    component.find('.closeIcon').simulate('click');

    expect(hideModal).toHaveBeenCalled();
  });

  test('mousedown event bubbling from the modal dialog is prevented', () => {
    const event = { stopPropagation: jest.fn() };
    component.find('.modal').simulate('mousedown', event);

    expect(event.stopPropagation).toHaveBeenCalled();
  });
});
