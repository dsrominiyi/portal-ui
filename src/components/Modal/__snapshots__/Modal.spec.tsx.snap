// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal hidden Modal matches snapshot 1`] = `
<Modal
  body={
    Array [
      "body",
    ]
  }
  footer="footer"
  header="header"
  hideModal={[MockFunction]}
  isVisible={false}
>
  <Portal>
    <Portal
      containerInfo={<div />}
    >
      <CSSTransition
        classNames={
          Object {
            "enter": "overlayEnter",
            "enterActive": "overlayEnterActive",
            "exit": "overlayExit",
            "exitActive": "overlayExitActive",
          }
        }
        in={false}
        mountOnEnter={true}
        timeout={300}
        unmountOnExit={true}
      >
        <Transition
          appear={false}
          enter={true}
          exit={true}
          in={false}
          mountOnEnter={true}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={300}
          unmountOnExit={true}
        />
      </CSSTransition>
    </Portal>
  </Portal>
</Modal>
`;

exports[`Modal visible Modal matches snapshot 1`] = `
<Modal
  body={
    Array [
      "body",
    ]
  }
  footer="footer"
  header="header"
  hideModal={[MockFunction]}
  isVisible={true}
>
  <Portal>
    <Portal
      containerInfo={
        <div>
          <div
            class="modalOverlay"
          >
            <div
              class="modal visible"
            >
              <div
                class="closeIcon"
              >
                <svg
                  height="20"
                  viewBox="0.5 0 20 20"
                  width="20"
                >
                  <path
                    d="M16 16L4 4M16 4L4 16"
                    fill="none"
                    stroke="#000"
                    stroke-width="1.06"
                  />
                </svg>
              </div>
              <div
                class="header"
              >
                header
              </div>
              <div
                class="body"
              >
                body
              </div>
              <div
                class="footer"
              >
                footer
              </div>
            </div>
          </div>
        </div>
      }
    >
      <CSSTransition
        classNames={
          Object {
            "enter": "overlayEnter",
            "enterActive": "overlayEnterActive",
            "exit": "overlayExit",
            "exitActive": "overlayExitActive",
          }
        }
        in={true}
        mountOnEnter={true}
        timeout={300}
        unmountOnExit={true}
      >
        <Transition
          appear={false}
          enter={true}
          exit={true}
          in={true}
          mountOnEnter={true}
          onEnter={[Function]}
          onEntered={[Function]}
          onEntering={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onExiting={[Function]}
          timeout={300}
          unmountOnExit={true}
        >
          <div
            className="modalOverlay"
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <div
              className="modal visible"
              onMouseDown={[Function]}
            >
              <div
                className="closeIcon"
                onClick={[MockFunction]}
              >
                <SvgClose
                  viewBox="0.5 0 20 20"
                >
                  <svg
                    height={20}
                    viewBox="0.5 0 20 20"
                    width={20}
                  >
                    <path
                      d="M16 16L4 4M16 4L4 16"
                      fill="none"
                      stroke="#000"
                      strokeWidth={1.06}
                    />
                  </svg>
                </SvgClose>
              </div>
              <div
                className="header"
              >
                header
              </div>
              <div
                className="body"
              >
                body
              </div>
              <div
                className="footer"
              >
                footer
              </div>
            </div>
          </div>
        </Transition>
      </CSSTransition>
    </Portal>
  </Portal>
</Modal>
`;
